### DISCUSSION TITLE
Integration with /etc/passwd in Linux

### DISCUSSION THREAD

**Discussion by ghost** [MarcJHuber/event-driven-servers#75] [Q&A]
Hello. 

I want to add authentication in tac_plus service via local username and password of OS Linux Ubuntu. How can I mention that my password for users should be taken from file /etc/passwd in tac_plus.cfg config file? Can someone give some advice and help?

        user = test {
                login = clear MyOwnPassword
                member = admin
                service = shell {
                        default command = permit
                        default attribute = permit
                        set priv-lvl = 15
 
 Or maybe I have a way to crypt passwords in my config file? The syntax below didn't work 

 user = test {
                login = crypt 'MyOwnPassword'

**Comment by MarcJHuber** [comment]
Hi,

/etc/password on Linux doesn't come with passwords, /etc/shadow does.

You can try the "pammavis" backend:

````
id = tac_plus {
...
    mavis module = external { exec = /usr/local/sbin/pammavis }
...
    login backend = mavis
...
    user = test {
        login = mavis
....
    }
}
````

The "shadow" Perl module can handle this too. You can find quite some information about the supported authentication mechanisms in the "mavis" docs, online available at

[https://www.pro-bono-publico.de/projects/mavis.html](https://www.pro-bono-publico.de/projects/mavis.html)

Please keep in mind that tac_plus is deprecated and tac_plus-ng is recommended, especially for new installations.

Regarding "login = crypt ..." not working: tac_plus can cope with whatever password hashes your operating system supports. Please double-check that the hash you're using isn't malformed.

Cheers,

Marc

**Comment by MarcJHuber** [comment]
Closed.

**Comment by MarcJHuber** [comment]
.
